src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -157,7 +157,7 @@ public TestPlatform()
         }
 
         /// <inheritdoc/>
-        public void StartTestSession(
+        public bool StartTestSession(
```

Comment:
Wouldn't that make the code (well, in this case documentation) harder to maintain ? Whenever there's a change in the implementation of `StartTestSession` or any of the methods it calls, for example when adding better error handling conditions in any of the called methods, we'd have to remember to update the documentation to mention that case too.

The documentation will easily get out of sync in this way and it may even be confusing to the reader who may assume the error cases enumerated there are the only ones where `StartTestSession` would return false. I think it'd be better to keep it more generic as it is now and rely on TP logging to truly figure out what went wrong.

Reply:
If this is a public interface then wouldn't it be a breaking change to limit the set of cases where the sessions cannot be created?

The idea for documentation was to ensure that there exists a contract between TE and TP on what conditions should be met to create sessions. So that it doesn't happen that the way we're currently creating sessions would suddenly become invalid.

However I agree that if suddenly sessions would support more scenarios (like in proc vstest.console) keeping the documentation up to date might be more difficult.

Reply:
I get what you meant now and I agree it's usually a good idea to have those cases documented somewhere, but I'm reluctant to commit to such a detailed contract between TP and TE because that will mean we'll have to maintain those cases forever, even if down the road we find some bugs that invalidate some cases in which test session creation used to succeed.

Also this exposes a great deal of implementation details to the caller that I'm not really sure is benefic to anyone. TP will have its hands tied in the ways it can change the underlying implementation and TE will probably make much more assumptions about the way test session creation works than it needs to. I still think the public API should be enough of a contract between them (i.e. the method name, the parameters and return value).

As a closing note I think TE should have a fallback implementation if test session creation fails (i.e. the old way of doing things) and then those failure cases won't really be relevant anymore.